pkg/webhookconfig/common.go

```diff
@@ -92,16 +92,17 @@ func (wrc *Register) GetKubePolicyClusterRoleName() (*unstructured.Unstructured,
 // GetKubePolicyDeployment gets Kyverno deployment using the resource cache
 // it does not initialize any client call
 func (wrc *Register) GetKubePolicyDeployment() (*apps.Deployment, *unstructured.Unstructured, error) {
-	lister, _ := wrc.resCache.GetGVRCache("Deployment")
-	kubePolicyDeployment, err := lister.NamespacedLister(config.KyvernoNamespace).Get(config.KyvernoDeploymentName)
+	deploy, err := wrc.kDeplLister.Deployments(config.KyvernoNamespace).Get(config.KyvernoDeploymentName)
 	if err != nil {
 		return nil, nil, err
 	}
-	deploy := apps.Deployment{}
-	if err = runtime.DefaultUnstructuredConverter.FromUnstructured(kubePolicyDeployment.UnstructuredContent(), &deploy); err != nil {
-		return nil, kubePolicyDeployment, err
+	kubePolicyDeployment := unstructured.Unstructured{}
+	content, err := runtime.DefaultUnstructuredConverter.ToUnstructured(deploy)
```

Comment:
This method previously led to a severe memory issue, see https://github.com/realshuting/kyverno/commit/0f0c070072f3c678aa33bc0f280bf1b31c13d7e8#diff-06d6ce2592c9ed3bf6917bff69a95dcf313bc52a3b367e6e2a66c5cf87b4f68aL110. 

Please use JSON marshaler/unmarshaler for type conversion.